pkgs/applications/science/logic/potassco/clingcon.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, cmake, bison, re2c }:
+
+stdenv.mkDerivation rec {
+  pname = "clingcon";
+  version = "3.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "potassco";
+    repo = "${pname}";
+    rev = "d36fd87c8937b0ae2b39593aab26f3eb717749d3";
+    fetchSubmodules = true;
+    sha256 = "1q7517h10jfvjdk2czq8d6y57r8kr1j1jj2k2ip2qxkpyfigk4rs";
+   };
+
+  postPatch = ''
+    # deal with clingcon through git submodules recursively importing
+    # an outdated version of libpotassco which uses deprecated <xlocale.h> header in .cpp files
+    find ./ -type f -exec sed -i 's/<xlocale.h>/<locale.h>/g' {} \;
+  '';
+
+  nativeBuildInputs = [ cmake bison re2c ];
+
+  cmakeFlags = [
+    "-DCLINGCON_MANAGE_RPATH=ON"
+    "-DCLINGO_BUILD_WITH_PYTHON=OFF"
```

Comment:
These seem like interesting features! Did you try to build it with them? If you did, it's common in Nixpkgs to add an argument to a `callPackage` such as `luaSupport` or `pythonSupport`. Try to get a hint of how it is done by looking at (e.g) https://github.com/NixOS/nixpkgs/blob/1ddb140d95c6a1ab914f1d9f3d5169b12074c333/pkgs/tools/security/sequoia/default.nix

There are other good or better examples out there, you can `git grep pythonSupport` or `git grep `enablePython` if you'd like to get more inspiration.

Reply:
Seems like [clingcon's next version](https://github.com/potassco/clingcon/blob/wip/CHANGES.md) will drop support for lua and python bindings. Maybe it's not worth the effort, as @7c6f434c suggests, to add it now.

Reply:
Makes sense, mark this and https://github.com/NixOS/nixpkgs/pull/80719/files#r382840015 as resolved.